sdk/common/src/main/scala/org/typelevel/otel4s/sdk/context/propagation/TextMapPropagator.scala

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2023 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s
+package sdk
+package context
+package propagation
+
+/** The process of propagating data across process boundaries involves injecting
+  * and extracting values in the form of text into carriers that travel in-band.
+  *
+  * The encoding used for this process is expected to conform to HTTP Header
+  * Field semantics, and values are commonly encoded as request headers for
+  * RPC/HTTP requests.
+  *
+  * The carriers used for propagating the data are typically HTTP requests, and
+  * the process is often implemented using library-specific request
+  * interceptors. On the client side, values are injected into the carriers,
+  * while on the server side, values are extracted from them.
+  */
+trait TextMapPropagator {
```

Comment:
Nearly identical to the `TextMapPropagator` in the core module. 

Here I dropped the `F` constraint and `def inject[A: TextMapSetter](ctx: Vault, carrier: A): F[Unit]`.